src/main/java/jp/classmethod/aws/gradle/lambda/AWSLambdaInvokeTask.java

```diff
@@ -105,12 +105,14 @@ private void setupPayload(InvokeRequest request) throws IOException {
 		if (payload instanceof File) {
 			File file = (File) payload;
 			str = Files.toString(file, Charsets.UTF_8);
+			request.setPayload(str);
 		} else if (payload instanceof Closure) {
 			Closure<?> closure = (Closure<?>) payload;
 			str = closure.call().toString();
-		} else {
+			request.setPayload(str);
+		} else if (payload instanceof String) {
```

Comment:
@tellary I think the code below addresses your points (though I don't understand the first part of your second point) and still resolves the NPE problem.

if (payload instanceof File) {
  File file = (File) payload;
  str = Files.toString(file, Charsets.UTF_8);
  request.setPayload(str);
} else if (payload instanceof Closure) {
  Closure<?> closure = (Closure<?>) payload;
  str = closure.call().toString();
  request.setPayload(str);
} else if (payload instanceof String) {
  str = payload;
  request.setPayload(str);
} else if (payload) {
  str = payload.toString();
  request.setPayload(str);
}